### DISCUSSION TITLE
[UX]Consistency of Tools' Active and Deactivated behavior depending on what is selected

### DISCUSSION THREAD

**Discussion by Reqrefusion** [FreeCAD/FreeCAD#14291] [Discussions that will turn in to issues but need refining]
For many tools to work, something must be selected or something must be on the screen, but this behavior is not consistent. This makes it difficult to use. 

I will mostly give examples from the Technical Drawing workbench, but I want to start from partdesign to explain this behavior. 

Here, something must be selected for the Clone tool to be active. This is logical and should happen. Likewise, it is necessary to select something in the MapSketch tool. PartDesign and Sketcher workbenches are luckier.

However, the Technical Drawing workbench is a complete mess. By redirecting it nicely at first, all tools are disabled until anything is added to the page. However, as soon as something is added to the page, they turn on like Christmas lights. Although all tools except the smart measurement tool require selection of something. Forget about choosing something, very special things need to be chosen.

For example, when I select a line or point, the tool icons used to show the area still appear active. Or how do I angle an area? Or the leaderline option is not active when I select only view. Why can I use the ExtensionDrawCosmCircle3Points tool, which requires 3 points to draw, without selecting three points? 

The error messages I receive when I press these tools are incompatible with each other. The titles of the error messages vary between "Incorrect selection", the short name of the tool, and the long name. 

I generally gave examples from technical drawings because they can be noticed without going into depth. But other workbenches have similar UX issues.

**Comment by maxwxyz** [comment]
That's fairly generic, I'll transfer this to a discussion. Individual issues should be filed for single problems.

**Reply by Reqrefusion** [reply]
@maxwxyz Do I need to open an issue for all of them one by one? Why not, if you guarantee he won't mark it as spam.
I mean for each tool.

**Reply by maxwxyz** [reply]
no please not for every command. But different WB work differently here, I think we should define a common behavior. FEM WB is a great example where selecting things with an open command is different for almost every command.

**Reply by Reqrefusion** [reply]
> no please not for every command. But different WB work differently here, I think we should define a common behavior. FEM WB is a great example where selecting things with an open command is different for almost every command.

Oh, all my dreams have failed. Here, I am generally referring to the Technical Drawing workbench. I think I can open an issue here without making many changes. 

And yes, this is a very basic UX behavior: a tool is only active when it can run. To achieve this, all vehicles that give error messages should be examined and it should be ensured that they cannot be selected instead of the error message.

The first stage is the simplest, its activation regardless of choice. The second stage is its activation according to the appropriate selection. I believe that the first stage can be easily handled.
